src/graphics/reproject-texture.js

```diff
@@ -78,6 +78,31 @@ const packFloat32ToRGBA8 = (value, array, offset) => {
     }
 };
 
+// pack samples into texture-ready format
+const packSamples = (samples) => {
+    const numSamples = samples.length;
+
+    const w = Math.min(numSamples, 512);
+    const h = Math.ceil(numSamples / w);
+    const data = new Uint8Array(w * h * 4);
+
+    // normalize float data and pack into rgba8
+    let off = 0;
+    for (let i = 0; i < numSamples; ++i) {
+        packFloat32ToRGBA8(samples[i * 4 + 0] * 0.5 + 0.5, data, off + 0);
+        packFloat32ToRGBA8(samples[i * 4 + 1] * 0.5 + 0.5, data, off + 4);
+        packFloat32ToRGBA8(samples[i * 4 + 2] * 0.5 + 0.5, data, off + 8);
+        packFloat32ToRGBA8(samples[i * 4 + 3] / 8, data, off + 12);
```

Comment:
Removing four multiplications and the `/ 8` division each iteration (but I didn't test actual performance difference nor the code itself):

```suggestion
    const n = numSamples * 4;
    let i = 0;
    while (i < n) {
        packFloat32ToRGBA8(samples[i++] * 0.5 + 0.5, data, off + 0);
        packFloat32ToRGBA8(samples[i++] * 0.5 + 0.5, data, off + 4);
        packFloat32ToRGBA8(samples[i++] * 0.5 + 0.5, data, off + 8);
        packFloat32ToRGBA8(samples[i++] * 0.125, data, off + 12);
```

Reply:
thanks for the suggestion @kungfooman. you make good points ab performance. in this case i'm probably more concerned with readability than execution speed, so would prefer to keep it a little verbose.